app/models/manageiq/providers/redhat/infra_manager/vm.rb

```diff
@@ -28,6 +28,42 @@ def parent_cluster
   alias_method :owning_cluster, :parent_cluster
   alias_method :ems_cluster, :parent_cluster
 
+  def self.create_from_event(event)
```

Comment:
Creating the VM should really be done in the `Refresher`, ideally we can just create the hash and let `save_vm_inventory` handle creating the vm record, then queue a targeted refresh for the VM to link up the relationships to host/cluster/etc...